Rocket.Chat/Controllers/Auth/ConnectServerViewController.swift

```diff
@@ -163,39 +165,27 @@ final class ConnectServerViewController: BaseViewController {
         }
     }
 
-    func validate(url: URL, completion: @escaping RequestCompletion) {
-        let request = URLRequest(url: url)
-        let session = URLSession.shared
-
-        let task = session.dataTask(with: request, completionHandler: { (data, _, _) in
-            if let data = data {
-                guard let json = try? JSON(data: data) else { return completion(nil, true) }
-                Log.debug(json.rawString())
-
-                guard let version = json["version"].string else {
-                    return completion(nil, true)
-                }
+    func validate(completion: @escaping RequestCompletion) {
+        API.shared.fetch { (result: InfoResult?) -> Void in
```

Comment:
I'm not sure I understand how this works... `API.shared.fetch` will hit `/api/info` because the `T` is `InfoResult`? I don't feel like this is flexible and easy to understand.